src/main/webapp/groups.js

```diff
@@ -12,6 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** A function that adds event listeners to a DOM objects. */
+function addEventListeners() {
+    document.getElementById("search-groups-form").addEventListener('submit', event => {
+        event.preventDefault();
+        getSearchGroupResults();
+    });
+}
+
+/** Gets the name of the group the user search for and display the search results. */
+function redirectToResultsGroups() {
+    return redirectToResultsGroupsHelper(document, window);
+}
+
+/** Helper function for redirectToResultsGroups, used for testing purposes. */
+function redirectToResultsGroupsHelper(document, window) {
+    var group = document.getElementById("search-box-results").value;
+    
+    var url = "groups.html?group=" + encodeURIComponent(group);
+    window.location.href = url;
+
+    return false;
```

Comment:
Why do you need to return false here?

Reply:
Returning false cancels the default submission behavior which is to reload the page.